### PR TITLE
Updating Sockpeer service.

### PR DESCRIPTION
**Pull request by rahuldimri**

### CODE REVIEW

**Comment by mateuszrzeszutek** [comment]
Hey @rahuldimri ,
Can you make sure the code is properly formatted and add some tests covering the new path?

**Comment by rahuldimri** [comment]
Working on this.

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java [hunk 1]
```diff
@@ -61,8 +61,14 @@ public void onEnd(
...
     if (peerService != null) {
       attributes.put(SemanticAttributes.PEER_SERVICE, peerService);
+    } else {
+      // try map sockPeerName only if there was no match for peerName
+      String sockPeerName = attributesGetter.sockPeerName(request, response);
+      String sockPeerService = mapToPeerService(sockPeerName);
+      if (sockPeerService != null) {
+        attributes.put(SemanticAttributes.PEER_SERVICE, sockPeerService);
+      }
     }
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java:71]: Nit: you could restructure this code a little a bit and have just one `attributes.put` call:
```suggestion
    if (peerService == null) {
      String sockPeerName = attributesGetter.sockPeerName(request, response);
      peerService = mapToPeerService(sockPeerName);
    }
    if (peerService != null) {
      attributes.put(SemanticAttributes.PEER_SERVICE, peerService);
    }
```

**Review by mateuszrzeszutek — DISMISSED** [review]

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java [hunk 2]
```diff
@@ -61,8 +61,14 @@ public void onEnd(
...
-  }
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java:65]: I think you need to revert this change

Comment (rahuldimri) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java:65]: @mateuszrzeszutek thanks for the help. Do you mean I need to cancel this change ?

Comment (mateuszrzeszutek) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java:65]: No, just bring that one `}` line back

**Review by rahuldimri — COMMENTED** [review]
Closing the }

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java [hunk 3]
```diff
@@ -59,6 +59,10 @@ public void onEnd(
 
     String peerName = attributesGetter.getPeerName(request);
     String peerService = mapToPeerService(peerName);
+    if (peerService == null) {
+      String sockPeerName = attributesGetter.sockPeerName(request, response);
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java:63]:
```suggestion
      String sockPeerName = attributesGetter.getSockPeerName(request, response);
```

File: instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractorTest.java [hunk 4]
```diff
@@ -94,4 +96,57 @@ void shouldSetPeerNameIfItMatches() {
...
+  void shouldSetSockPeerNameIfItMatchesAndNoPeerNameProvided() {
+    // given
+    Map<String, String> peerServiceMapping = new HashMap<>();
+    peerServiceMapping.put("example.com", "myService");
+    peerServiceMapping.put("1.2.3.4", "someOtherService");
+
+    PeerServiceAttributesExtractor<String, String> underTest =
+        new PeerServiceAttributesExtractor<>(netAttributesExtractor, peerServiceMapping);
+
+    when(netAttributesExtractor.sockPeerName(any(), any())).thenReturn("example.com");
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractorTest.java:110]:
```suggestion
    when(netAttributesExtractor.getSockPeerName(any(), any())).thenReturn("example.com");
```

File: instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractorTest.java [hunk 5]
```diff
@@ -94,4 +96,57 @@ void shouldSetPeerNameIfItMatches() {
...
+    underTest.onEnd(endAttributes, context, "request", "response", null);
+
+    // then
+    assertThat(startAttributes.build()).isEmpty();
+    assertThat(endAttributes.build())
+        .containsOnly(entry(SemanticAttributes.PEER_SERVICE, "myService"));
+  }
+
+  @Test
+  void shouldSetNoSockPeerNameIfPeerNameMatches() {
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractorTest.java:127]: This test is almost the same as `shouldSetPeerNameIfItMatches`; instead of duplicating it, can you add the final `verify()` to that test case?

File: instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractorTest.java [hunk 6]
```diff
@@ -94,4 +96,57 @@ void shouldSetPeerNameIfItMatches() {
...
+    AttributesBuilder startAttributes = Attributes.builder();
+    underTest.onStart(startAttributes, context, "request");
+    AttributesBuilder endAttributes = Attributes.builder();
+    underTest.onEnd(endAttributes, context, "request", "response", null);
+
+    // then
+    assertThat(startAttributes.build()).isEmpty();
+    assertThat(endAttributes.build())
+        .containsOnly(entry(SemanticAttributes.PEER_SERVICE, "myService"));
+    verify(netAttributesExtractor, never()).sockPeerName(any(), any());
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractorTest.java:150]:
```suggestion
    verify(netAttributesExtractor, never()).getSockPeerName(any(), any());
```

**Review by mateuszrzeszutek — APPROVED** [review]
Thanks @rahuldimri !

**Comment by rahuldimri** [comment]
@mateuszrzeszutek thanks for your support in this PR. Do you mind to add labels on this issue ?.

**Comment by mateuszrzeszutek** [comment]
I'm not sure I'm following; what labels do you have in mind?

**Comment by trask** [comment]
hey @rahuldimri, could you elaborate on why you needed this? Is `net.peer.name` not being populated by some instrumentation? thx!

**Comment by trask** [comment]
@rahuldimri fyi, I suspect the need for this will go away with https://github.com/open-telemetry/opentelemetry-specification/pull/3402, where it's clarified that we should be populating (the equivalent of) `net.peer.name`, and we should only be populating (the equivalent of) `net.sock.peer.name` when going through an intermediary (e.g. proxy server)
